conda_mirror/conda_mirror.py

```diff
@@ -170,16 +175,92 @@ def pdb_hook(exctype, value, traceback):
          blacklist, whitelist)
 
 
-def download(url, filename, chunk_size=None):
+def _remove_package(pkg_path):
+    """
+    Log and remove a package.
+
+    Parameters
+    ----------
+    pkg_path : str
+        Path to a conda package that should be removed
+    """
+    logger.info("Removing: {}".format(pkg_path))
+    os.remove(pkg_path)
+
+
+def _validate(filename, md5, sha256, size):
+    try:
+        t = tarfile.open(filename)
+        index_json = t.extractfile('info/index.json').read().decode('utf-8')
+    except tarfile.TarError:
+        logging.debug("tarfile error encountered. Original error below.")
+        logging.debug(pformat(traceback.format_exc()))
+        logging.info("Removing package: %s", filename)
```

Comment:
Nit: Maybe use warn level? A corrupted tarball might indicate a bigger problem we need to investigate.